homeassistant/components/teslemetry/climate.py

```diff
@@ -153,3 +170,121 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
             self._attr_hvac_mode = HVACMode.HEAT_COOL
         self.async_write_ha_state()
+
+
+COP_MODES = {
+    "Off": HVACMode.OFF,
+    "On": HVACMode.COOL,
+    "FanOnly": HVACMode.FAN_ONLY,
+}
+
+COP_LEVELS = {
+    "Low": 30,
+    "Medium": 35,
+    "High": 40,
+}
+
+
+class TeslemetryCabinOverheatProtectionEntity(TeslemetryVehicleEntity, ClimateEntity):
+    """Telemetry vehicle cabin overheat protection entity."""
+
+    _attr_precision = PRECISION_WHOLE
+    _attr_target_temperature_step = 5
+    _attr_min_temp = 30
+    _attr_max_temp = 40
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = list(COP_MODES.values())
+    _enable_turn_on_off_backwards_compatibility = False
+    _attr_entity_registry_enabled_default = False
+
+    def __init__(
+        self,
+        data: TeslemetryVehicleData,
+        scopes: Scope,
+    ) -> None:
+        """Initialize the climate."""
+
+        super().__init__(data, "climate_state_cabin_overheat_protection")
+
+        # Supported Features
+        self._attr_supported_features = (
+            ClimateEntityFeature.TURN_ON | ClimateEntityFeature.TURN_OFF
+        )
+        if self.get("vehicle_config_cop_user_set_temp_supported"):
+            self._attr_supported_features |= ClimateEntityFeature.TARGET_TEMPERATURE
+
+        # Scopes
+        self.scoped = Scope.VEHICLE_CMDS in scopes
+        if not self.scoped:
+            self._attr_supported_features = ClimateEntityFeature(0)
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+
+        if (state := self.get("climate_state_cabin_overheat_protection")) is None:
+            self._attr_hvac_mode = None
+        else:
+            self._attr_hvac_mode = COP_MODES.get(state)
+
+        if (level := self.get("climate_state_cop_activation_temperature")) is None:
+            self._attr_target_temperature = None
+        else:
+            self._attr_target_temperature = COP_LEVELS.get(level)
+
+        self._attr_current_temperature = self.get("climate_state_inside_temp")
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+        await self.async_set_hvac_mode(HVACMode.COOL)
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+        await self.async_set_hvac_mode(HVACMode.OFF)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the climate temperature."""
+        self.raise_for_scope()
+
+        if temp := kwargs.get(ATTR_TEMPERATURE):
+            if temp == 30:
+                cop_mode = CabinOverheatProtectionTemp.LOW
+            elif temp == 35:
+                cop_mode = CabinOverheatProtectionTemp.MEDIUM
+            elif temp == 40:
+                cop_mode = CabinOverheatProtectionTemp.HIGH
+            else:
+                raise ServiceValidationError(
+                    "Cabin overheat protection does not support that temperature",
```

Comment:
```suggestion
```
the translations helper will return the error so no need to explicitly provide one if same as in translations